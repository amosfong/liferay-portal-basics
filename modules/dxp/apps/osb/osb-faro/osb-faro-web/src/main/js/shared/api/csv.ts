import sendRequest from 'shared/util/request';
import {CSVType} from 'shared/components/download-report/utils';

export const fetchCSV = (url: string) => fetch(url);

export const fetchCount = ({
	groupId,
	type,
	...data
}: {
	assetId?: string;
	assetType?: string;
	channelId: string;
	fromDate?: string;
	groupId: string;
	individualId?: string;
	rangeKey?: string;
	toDate?: string;
	type: CSVType;
}) =>
	sendRequest({
		data,
		method: 'GET',
		path: `main/${groupId}/reports/export/csv/${type}/count`
	});
