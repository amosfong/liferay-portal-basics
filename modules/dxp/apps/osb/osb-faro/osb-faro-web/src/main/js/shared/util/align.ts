export const POSITIONS = {
	BottomCenter: 4,
	BottomLeft: 5,
	BottomRight: 3,
	LeftCenter: 6,
	RightCenter: 2,
	TopCenter: 0,
	TopLeft: 7,
	TopRight: 1
};

export type Position = typeof POSITIONS[keyof typeof POSITIONS];

export function align(element, alignElement, position, autoBestAlign = true) {
	let bestRegion;

	if (autoBestAlign) {
		const suggestion = suggestAlignBestRegion(
			element,
			alignElement,
			position
		);
		position = suggestion.position;
		bestRegion = suggestion.region;
	} else {
		bestRegion = getAlignRegion(element, alignElement, position);
	}

	const computedStyle = window.getComputedStyle(element, null);

	if (computedStyle.getPropertyValue('position') !== 'fixed') {
		bestRegion.top += window.scrollY;
		bestRegion.left += window.scrollX;

		let offsetParent = element;
		while ((offsetParent = offsetParent.offsetParent)) {
			bestRegion.top -= offsetParent.offsetTop;
			bestRegion.left -= offsetParent.offsetLeft;
		}
	}

	element.style.top = `${bestRegion.top}px`;
	element.style.left = `${bestRegion.left}px`;

	return position;
}

function getAlignRegion(element, alignElement, position) {
	const r1 = alignElement.getBoundingClientRect();
	const r2 = element.getBoundingClientRect();
	let top = 0;
	let left = 0;

	switch (position) {
		case POSITIONS.TopCenter:
		default:
			top = r1.top - r2.height;
			left = r1.left + r1.width / 2 - r2.width / 2;
			break;
		case POSITIONS.RightCenter:
			top = r1.top + r1.height / 2 - r2.height / 2;
			left = r1.left + r1.width;
			break;
		case POSITIONS.BottomCenter:
			top = r1.bottom;
			left = r1.left + r1.width / 2 - r2.width / 2;
			break;
		case POSITIONS.LeftCenter:
			top = r1.top + r1.height / 2 - r2.height / 2;
			left = r1.left - r2.width;
			break;
		case POSITIONS.TopRight:
			top = r1.top - r2.height;
			left = r1.right - r2.width;
			break;
		case POSITIONS.BottomRight:
			top = r1.bottom;
			left = r1.right - r2.width;
			break;
		case POSITIONS.BottomLeft:
			top = r1.bottom;
			left = r1.left;
			break;
		case POSITIONS.TopLeft:
			top = r1.top - r2.height;
			left = r1.left;
			break;
	}

	return {
		bottom: top + r2.height,
		height: r2.height,
		left,
		right: left + r2.width,
		top,
		width: r2.width
	};
}

function suggestAlignBestRegion(element, alignElement, position) {
	let bestArea = 0;
	let bestPosition = position;
	let bestRegion = getAlignRegion(element, alignElement, bestPosition);
	let tryPosition = bestPosition;
	let tryRegion = bestRegion;

	const viewportRegion = getViewportRegion();

	for (let i = 0; i < 8; ) {
		if (intersectRegion(viewportRegion, tryRegion)) {
			const visibleRegion = intersection(viewportRegion, tryRegion);

			const area = visibleRegion.width * visibleRegion.height;

			if (area > bestArea) {
				bestArea = area;
				bestRegion = tryRegion;
				bestPosition = tryPosition;
			}

			if (insideRegion(viewportRegion, tryRegion)) {
				break;
			}
		}
		tryPosition = (position + ++i) % 8;
		tryRegion = getAlignRegion(element, alignElement, tryPosition);
	}

	return {
		position: bestPosition,
		region: bestRegion
	};
}

function getViewportRegion() {
	const height = document.documentElement.offsetHeight;
	const width = document.documentElement.offsetWidth;

	return makeRegion(height, height, 0, width, 0, width);
}

function insideRegion(r1, r2) {
	return (
		r2.top >= r1.top &&
		r2.bottom <= r1.bottom &&
		r2.right <= r1.right &&
		r2.left >= r1.left
	);
}

function intersectRect(x0, y0, x1, y1, x2, y2, x3, y3) {
	return !(x2 > x1 || x3 < x0 || y2 > y1 || y3 < y0);
}

function intersectRegion(r1, r2) {
	return intersectRect(
		r1.top,
		r1.left,
		r1.bottom,
		r1.right,
		r2.top,
		r2.left,
		r2.bottom,
		r2.right
	);
}

function intersection(r1, r2) {
	if (!intersectRegion(r1, r2)) {
		return null;
	}
	const bottom = Math.min(r1.bottom, r2.bottom);
	const right = Math.min(r1.right, r2.right);
	const left = Math.max(r1.left, r2.left);
	const top = Math.max(r1.top, r2.top);

	return makeRegion(bottom, bottom - top, left, right, top, right - left);
}

function makeRegion(bottom, height, left, right, top, width) {
	return {
		bottom,
		height,
		left,
		right,
		top,
		width
	};
}
