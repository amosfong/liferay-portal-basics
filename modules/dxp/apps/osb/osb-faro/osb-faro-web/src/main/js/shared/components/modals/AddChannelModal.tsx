import ClayButton from '@clayui/button';
import Form, {
	validateMaxLength,
	validateMinLength,
	validateRequired
} from 'shared/components/form';
import Loading, {Align} from 'shared/components/Loading';
import Modal from 'shared/components/modal';
import React from 'react';
import {Modal as ModalTypes} from 'shared/types';
import {sequence} from 'shared/util/promise';

interface IAddChannelModalProps {
	onClose: ModalTypes.close;
	onSubmit: (value: {name: string}) => void;
}

const AddChannelModal: React.FC<IAddChannelModalProps> = ({
	onClose,
	onSubmit
}) => (
	<Modal className='add-channel-modal'>
		<Form
			initialValues={{
				name: ''
			}}
			onSubmit={onSubmit}
		>
			{({handleSubmit, isSubmitting, isValid}) => (
				<Form.Form onSubmit={handleSubmit}>
					<Modal.Header
						onClose={onClose}
						title={Liferay.Language.get('new-property')}
					/>

					<Modal.Body>
						<Form.Input
							autoFocus
							label={Liferay.Language.get('property-name')}
							name='name'
							validate={sequence([
								validateRequired,
								validateMaxLength(65),
								validateMinLength(3)
							])}
						/>
					</Modal.Body>

					<Modal.Footer>
						<ClayButton
							className='button-root'
							displayType='secondary'
							onClick={onClose}
						>
							{Liferay.Language.get('cancel')}
						</ClayButton>

						<ClayButton
							className='button-root'
							disabled={isSubmitting || !isValid}
							displayType='primary'
							type='submit'
						>
							{isSubmitting && <Loading align={Align.Left} />}

							{Liferay.Language.get('save')}
						</ClayButton>
					</Modal.Footer>
				</Form.Form>
			)}
		</Form>
	</Modal>
);

export default AddChannelModal;
