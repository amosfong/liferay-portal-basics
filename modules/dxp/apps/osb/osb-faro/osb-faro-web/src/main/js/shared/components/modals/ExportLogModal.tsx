import ClayButton from '@clayui/button';
import DateRangeInput, {DateRange} from 'shared/components/DateRangeInput';
import Loading, {Align} from 'shared/components/Loading';
import Modal from 'shared/components/modal';
import moment from 'moment';
import React, {useState} from 'react';
import {downloadDataAsFile} from 'shared/util/util';

interface IExportLogModalProps {
	description: string;
	fileName: string;
	groupId: string;
	onClose: () => void;
	onSubmit: ({
		fromDate,
		toDate
	}: {
		fromDate: string;
		toDate: string;
	}) => Promise<any>;
	title: string;
}

const ExportLogModal: React.FC<IExportLogModalProps> = ({
	description,
	fileName,
	groupId,
	onClose,
	onSubmit,
	title
}) => {
	const [dateRange, setDateRange] = useState<DateRange>({
		end: null,
		start: null
	});

	const [loading, setLoading] = useState<boolean>(false);

	const isValid = (): boolean =>
		moment(dateRange.end).isValid() && moment(dateRange.start).isValid();

	const {end: toDate, start: fromDate} = dateRange;

	return (
		<Modal className='export-log-modal-root'>
			<Modal.Header onClose={onClose} title={title} />

			<Modal.Body>
				<p className='text-secondary'>{description}</p>

				<div className='h4'>
					{Liferay.Language.get('request-date-range')}
				</div>

				<div className='d-flex'>
					<DateRangeInput
						className='w-100'
						groupId={groupId}
						onChange={setDateRange}
						value={dateRange}
					/>

					<ClayButton
						className='button-root ml-2'
						disabled={!isValid()}
						displayType='primary'
						onClick={() => {
							setLoading(true);

							onSubmit({fromDate, toDate})
								.then((data: string) => {
									downloadDataAsFile({
										data,
										name: fileName,
										type: 'text/csv'
									});

									setLoading(false);

									onClose();
								})
								.catch(() => {
									setLoading(false);
								});
						}}
					>
						{loading && <Loading align={Align.Left} />}

						{Liferay.Language.get('download')}
					</ClayButton>
				</div>
			</Modal.Body>
		</Modal>
	);
};

export default ExportLogModal;
