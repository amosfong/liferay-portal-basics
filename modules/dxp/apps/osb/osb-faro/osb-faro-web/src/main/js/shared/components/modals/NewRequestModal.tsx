import * as API from 'shared/api';
import ClayButton from '@clayui/button';
import ClayLink from '@clayui/link';
import ClayMultiSelect from '@clayui/multi-select';
import FileDropTarget from 'shared/components/FileDropTarget';
import Form from 'shared/components/form';
import getCN from 'classnames';
import Input from 'shared/components/Input';
import Loading, {Align} from 'shared/components/Loading';
import Modal from 'shared/components/modal';
import React, {useEffect, useRef, useState} from 'react';
import {Formik, FormikValues} from 'formik';
import {NetworkStatus} from '@clayui/data-provider';
import {paginationDefaults} from 'shared/util/pagination';
import {sub} from 'shared/util/lang';
import {useDebounce} from 'shared/hooks/useDebounce';

const SAMPLE_CSV = 'user@example.com\nuser1@example.com\nuser2@example.com';

const {page} = paginationDefaults;

const AUTOCOMPLETE_DELTA = 5;

enum SubjectIdType {
	ByEmail = 'email',
	ByFile = 'file'
}

const getCheckboxLabel = (title: string, subtitle: string): React.ReactNode => (
	<span className='checkbox-label'>
		{title}

		<span className='text-secondary'>{` - ${subtitle}`}</span>
	</span>
);

interface INewRequestModalProps {
	groupId: string;
	onClose: () => void;
	onSubmit: (value: {
		emailAddresses?: string[];
		fileName?: string;
		types: string[];
	}) => void;
}

const NewRequestModal: React.FC<INewRequestModalProps> = ({
	groupId,
	onClose,
	onSubmit
}) => {
	const [items, setItems] = useState([]);
	const [fileName, setFileName] = useState(null);
	const [email, setEmail] = useState('');
	const [networkStatus, setNetworkStatus] = useState(NetworkStatus.Unused);
	const [emails, setEmails] = useState([]);

	const debouncedEmail = useDebounce(email, 500);

	useEffect(() => {
		setNetworkStatus(NetworkStatus.Loading);

		async function fetchIndividuals() {
			const {items} = await API.individuals.search({
				delta: AUTOCOMPLETE_DELTA,
				filter: debouncedEmail
					? `contains(demographics/email/value, '${debouncedEmail}')`
					: '',
				groupId,
				page
			});

			setItems(
				items.map(({id, properties: {email}}) => ({
					label: email,
					value: id
				}))
			);
			setNetworkStatus(NetworkStatus.Unused);
		}

		fetchIndividuals();
	}, [debouncedEmail]);

	const _formRef = useRef<Formik>();

	const handleAccessClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {setFieldValue} = _formRef.current;

			setFieldValue('accessRequest', checked);
		}
	};

	const handleDeleteClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {setFieldValue} = _formRef.current;

			setFieldValue('deleteRequest', checked);

			setFieldValue('suppressRequest', checked);
		}
	};

	const handleFileChange = file => {
		if (file) {
			const {completed, response, status} = file;

			const fileUploaded = completed && status !== 500;

			setFileName(fileUploaded ? response : null);
		} else {
			setFileName(null);
		}
	};

	const handleSubmit = ({
		accessRequest,
		deleteRequest,
		subjectIdType,
		suppressRequest
	}) => {
		let types = [];

		if (accessRequest) {
			types = [...types, 'ACCESS'];
		}

		if (suppressRequest) {
			types = [...types, 'SUPPRESS'];
		}

		if (deleteRequest) {
			types = [...types, 'DELETE'];
		}

		if (subjectIdType === SubjectIdType.ByEmail) {
			onSubmit({
				emailAddresses: emails.map(({label}) => label),
				types
			});
		} else {
			onSubmit({
				fileName,
				types
			});
		}
	};

	const handleSuppressClick = event => {
		const {checked} = event.target;

		if (_formRef.current) {
			const {
				setFieldValue,
				state: {
					values: {deleteRequest}
				}
			} = _formRef.current;

			if (!deleteRequest) {
				setFieldValue('suppressRequest', checked);
			}
		}
	};

	const isValid = ({
		accessRequest,
		deleteRequest,
		subjectIdType,
		suppressRequest
	}: FormikValues): boolean => {
		const requests = accessRequest || deleteRequest || suppressRequest;

		const byEmail =
			subjectIdType === SubjectIdType.ByEmail && emails.length;
		const byFile = subjectIdType === SubjectIdType.ByFile && fileName;

		return requests && (byEmail || byFile);
	};

	return (
		<Modal className='new-request-modal-root'>
			<Modal.Header
				onClose={onClose}
				title={Liferay.Language.get('new-request')}
			/>

			<Form
				initialValues={{
					accessRequest: false,
					deleteRequest: false,
					subjectIdType: SubjectIdType.ByEmail,
					suppressRequest: false
				}}
				onSubmit={handleSubmit}
				ref={_formRef}
			>
				{({handleSubmit, isSubmitting, values}) => (
					<Form.Form onSubmit={handleSubmit}>
						<Modal.Body>
							<p className='text-secondary'>
								{Liferay.Language.get(
									'new-requests-will-be-added-to-the-queue-and-you-will-be-notified-once-the-job-has-completed-running'
								)}
							</p>

							<Form.Group>
								<div className='h4'>
									{Liferay.Language.get('job-type')}
								</div>

								<Form.GroupItem>
									<Form.Checkbox
										autoFocus
										label={getCheckboxLabel(
											Liferay.Language.get('access'),
											Liferay.Language.get(
												'creates-downloadable-file-of-all-the-data-collected-related-to-a-user'
											)
										)}
										name='accessRequest'
										onChange={handleAccessClick}
										value='accessRequest'
									/>
								</Form.GroupItem>

								<Form.GroupItem>
									<Form.Checkbox
										label={getCheckboxLabel(
											Liferay.Language.get('delete'),
											Liferay.Language.get(
												'delete-users-pii-and-add-them-to-the-suppression-list'
											)
										)}
										name='deleteRequest'
										onChange={handleDeleteClick}
										value='deleteRequest'
									/>
								</Form.GroupItem>

								<Form.GroupItem>
									<Form.Checkbox
										disabled={values.deleteRequest}
										label={getCheckboxLabel(
											Liferay.Language.get('suppress'),
											Liferay.Language.get(
												'suppress-identity-resolution-of-users-based-on-their-email'
											)
										)}
										name='suppressRequest'
										onChange={handleSuppressClick}
										value='suppressRequest'
									/>
								</Form.GroupItem>
							</Form.Group>

							<Form.Group>
								<div className='h4'>
									{Liferay.Language.get('data-subject-id')}
								</div>

								<Form.GroupItem>
									<Form.RadioGroup name='subjectIdType'>
										<Form.RadioGroup.Option
											key={SubjectIdType.ByEmail}
											label={Liferay.Language.get(
												'find-by-email'
											)}
											value={SubjectIdType.ByEmail}
										/>

										<Form.RadioGroup.Subsection>
											<Input.Group>
												<ClayMultiSelect
													allowsCustomLabel={false}
													disabled={
														values.subjectIdType ===
														SubjectIdType.ByFile
													}
													loadingState={networkStatus}
													onChange={setEmail}
													onItemsChange={setEmails}
													placeholder={Liferay.Language.get(
														'example-email'
													)}
													sourceItems={items}
													value={email}
												/>
											</Input.Group>
										</Form.RadioGroup.Subsection>

										<Form.RadioGroup.Option
											key={SubjectIdType.ByFile}
											label={Liferay.Language.get(
												'upload-file'
											)}
											value={SubjectIdType.ByFile}
										/>

										<Form.RadioGroup.Subsection
											className={getCN({
												hide:
													values.subjectIdType ===
													SubjectIdType.ByEmail
											})}
										>
											<div>
												<FileDropTarget
													fileTypes={['.csv']}
													onChange={handleFileChange}
													uploadURL={`/o/proxy/download/data-control-tasks?projectGroupId=${groupId}`}
												/>

												<div className='example-file-text text-secondary'>
													{sub(
														Liferay.Language.get(
															'please-upload-files-in-csv-format.-a-sample-file-can-be-found-x'
														),
														[
															<ClayLink
																download='example_user_request.csv'
																// @ts-ignore
																externalLink
																href={`data:text/octet-stream;charset=utf-8,${SAMPLE_CSV}`}
																key='EXAMPLE_FILE'
															>
																{Liferay.Language.get(
																	'here-fragment'
																)}
															</ClayLink>
														],
														false
													)}
												</div>
											</div>
										</Form.RadioGroup.Subsection>
									</Form.RadioGroup>
								</Form.GroupItem>
							</Form.Group>
						</Modal.Body>

						<Modal.Footer>
							<ClayButton
								className='button-root'
								displayType='secondary'
								onClick={onClose}
							>
								{Liferay.Language.get('cancel')}
							</ClayButton>

							<ClayButton
								className='button-root'
								disabled={!isValid(values)}
								displayType='primary'
								type='submit'
							>
								{isSubmitting && <Loading align={Align.Left} />}

								{Liferay.Language.get('save')}
							</ClayButton>
						</Modal.Footer>
					</Form.Form>
				)}
			</Form>
		</Modal>
	);
};

export default NewRequestModal;
