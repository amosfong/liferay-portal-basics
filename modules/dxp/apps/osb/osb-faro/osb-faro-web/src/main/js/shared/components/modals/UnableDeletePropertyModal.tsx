import ClayButton from '@clayui/button';
import ClayLink from '@clayui/link';
import Modal from 'shared/components/modal';
import React from 'react';
import URLConstants from 'shared/util/url-constants';
import {sub} from 'shared/util/lang';

const UnableDeletePropertyModal: React.FC<{
	onClose: () => void;
}> = ({onClose}) => (
	<Modal type='danger'>
		<Modal.Header
			onClose={onClose}
			title={Liferay.Language.get('unable-to-delete-property')}
		/>

		<Modal.Body inlineScroller>
			<p className='text-secondary'>
				{sub(
					Liferay.Language.get(
						'ensure-no-sites-and-channels-are-assigned-to-it-before-deleting-a-property'
					),
					[
						<ClayLink
							href={URLConstants.DeletePropertyDocumentation}
							key='URL'
							target='_blank'
						>
							{Liferay.Language.get(
								'access-our-documentation-to-learn-more'
							)}
						</ClayLink>
					],
					false
				)}
			</p>
		</Modal.Body>

		<Modal.Footer>
			<ClayButton
				className='button-root'
				displayType='danger'
				onClick={onClose}
			>
				{Liferay.Language.get('done')}
			</ClayButton>
		</Modal.Footer>
	</Modal>
);

export default UnableDeletePropertyModal;
