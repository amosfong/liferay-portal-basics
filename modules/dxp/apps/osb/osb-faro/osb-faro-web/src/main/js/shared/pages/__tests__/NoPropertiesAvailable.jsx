jest.mock('shared/actions/modals', () => ({
	actionTypes: {},
	close: jest.fn(),
	modalTypes: {},
	open: jest.fn(() => ({meta: {}, payload: {}, type: 'open'}))
}));

import * as API from 'shared/api';
import mockStore from 'test/mock-store';
import NoPropertiesAvailable from '../NoPropertiesAvailable';
import React from 'react';
import {cleanup, fireEvent, render} from '@testing-library/react';
import {getImmutableMock, mockMemberUser, mockUser} from 'test/data';
import {open} from 'shared/actions/modals';
import {Provider} from 'react-redux';
import {StaticRouter} from 'react-router';
import {User} from 'shared/util/records';
import {waitForLoadingToBeRemoved} from 'test/helpers';

jest.unmock('react-dom');

const DefaultComponent = props => (
	<Provider store={mockStore()}>
		<StaticRouter>
			<NoPropertiesAvailable
				currentUser={getImmutableMock(User, mockUser)}
				groupId='123123'
				{...props}
			/>
		</StaticRouter>
	</Provider>
);

describe('NoPropertiesAvailable', () => {
	afterEach(cleanup);

	it('should render', async () => {
		API.dataSource.search.mockReturnValueOnce(Promise.resolve({total: 0}));

		const {container} = render(<DefaultComponent />);

		jest.runAllTimers();

		await waitForLoadingToBeRemoved(container);

		expect(container).toMatchSnapshot();
	});

	it('should render w/ Create Property button', async () => {
		const {container, queryByText} = render(<DefaultComponent />);

		jest.runAllTimers();

		await waitForLoadingToBeRemoved(container);

		expect(queryByText('Create Property')).toBeTruthy();
	});

	it('should not display the "Start" button to trigger onboarding if the user is not an AC admin', () => {
		const {queryByRole} = render(
			<DefaultComponent
				currentUser={getImmutableMock(User, mockMemberUser)}
			/>
		);

		jest.runAllTimers();

		expect(queryByRole('button')).toBeNull();
	});

	it('should call open on "Start" click', async () => {
		API.dataSource.search.mockReturnValueOnce(Promise.resolve({total: 0}));

		const {container, queryByText} = render(
			<DefaultComponent open={open} />
		);

		expect(open).not.toBeCalled();

		jest.runAllTimers();

		await waitForLoadingToBeRemoved(container);

		fireEvent.click(queryByText('Start'));

		expect(open).toBeCalled();
	});
});
