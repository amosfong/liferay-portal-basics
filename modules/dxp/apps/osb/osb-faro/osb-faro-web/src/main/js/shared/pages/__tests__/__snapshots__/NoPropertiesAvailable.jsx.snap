// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoPropertiesAvailable should render 1`] = `
<div>
  <div
    class="index-root no-properties-available-root"
  >
    <div
      class="maintenance-alert-root"
    />
    <header
      class="header-root"
    >
      <div
        class="header-container"
      >
        <div
          class="row-root align-items-center d-flex justify-content-between"
        >
          <ol
            class="breadcrumb breadcrumbs-root"
          />
        </div>
        <div
          class="header-section title-section"
        >
          <span
            class="align-items-center d-flex"
          >
            <h1
              class="title text-truncate"
            >
              <span
                class="text-truncate"
                data-tooltip="false"
                title=""
              >
                No Properties Available
              </span>
            </h1>
          </span>
        </div>
      </div>
    </header>
    <div
      class="body-root page-container"
    >
      <span
        class="children-wrapper"
      >
        <div
          class="no-results-root display-card flex-grow-1"
        >
          <div
            class="no-results-content spacer"
          >
            <div
              class="no-results-icon"
            >
              <svg
                class="lexicon-icon lexicon-icon-ac-satellite icon-root icon-size-xxxl"
                role="presentation"
              >
                <use
                  href="#ac-satellite"
                />
              </svg>
            </div>
            <div
              class="h4 no-results-title"
            >
              You are Almost There
            </div>
            <div
              class="no-results-description"
            >
              <p>
                First, complete the onboarding to get your workspace set up.
              </p>
              <button
                class="button-root btn btn-primary"
                type="button"
              >
                Start
              </button>
            </div>
          </div>
        </div>
      </span>
    </div>
  </div>
</div>
`;
